lib/compress.js

```diff
@@ -1990,6 +1990,15 @@ merge(Compressor.prototype, {
         return this.pure = pure;
     });
 
+    var global_pure_fns = makePredicate("Boolean decodeURI decodeURIComponent Date encodeURI encodeURIComponent Error escape EvalError isFinite isNaN Number Object parseFloat parseInt RangeError ReferenceError String SyntaxError TypeError unescape URIError");
```

Comment:
Isn't `Date()` subject to the locale for the machine running uglify? Hmm - I guess for the purpose of determining whether it is a pure function it doesn't matter.

Reply:
This optimisation only kicks in when the return value is discarded, so I think we should only concern about side effects.

Reply:
The name is fine. 👍